setup.md

```diff
@@ -3,9 +3,27 @@
 - At the root, it has the structure of a foundry repo.
 - All circuit specific code is in [./circuits](./circuits) directory. Circuits are written in circom.
 
+## Prerequisite installations
+- Install custom circom fork
+  -  We are using Nova Scotia which needs a custom Circom fork: https://github.com/nalinbhardwaj/circom/tree/pasta
+  - Checkout the `pasta` branch locally and follow https://docs.circom.io/getting-started/installation/.
+  - This replaces your global circom binary.
+
+## Smart contracts
+
 To run and test smart contracts:
 - forge test
 
-For circuits:
+## Circuits
 - Run `yarn install` in [./circuits/circuits](./circuits/circuits) and [./circuits/scripts](./circuits/scripts).
-- Navigate to [./circuits/scripts](./circuits/scripts) and run `gen_keys.sh`.
\ No newline at end of file
+- Install git submodule [circom-ecdsa@v0.0.1](https://github.com/0xPARC/circom-ecdsa/releases/tag/v0.0.1).
+- Navigate to [./circuits/circuits/git_modules/circom-ecdsa](./circuits/circuits/git_modules/circom-ecdsa), and run `ln -s  ../../node_modules node_modules`.
+  - Make sure to not commit the updated submodule.
+- Navigate to [./circuits/scripts](./circuits/scripts), set `CIRCUIT_NAME` to the name of the circuit you want to compile.
```

Comment:
This could be an argument to the script, so that we can do a simple one liner like `gen_keys.sh mint` or `gen_keys.sh mint.circom`.

Reply:
perhaps can be done in a future PR?